git/Contributing.md

```diff
@@ -0,0 +1,58 @@
+
+## Branches
+
+One of the most powerful features of Git development flow are **branches**. After repository is created, `master` represents predefined branch. This branch is defined for _production_ version of application. Further more, it should be **protected** for force pushes (`git push origin master -f`).
+
+Another important branch is `dev`. Recommended way for creating `dev` branch is typing following command:
+```
+git checkout dev
+```
+
+`dev` branch serves for holding finished **tasks** which need additional testing before **pushing** them to **production**. In practice, that branch is used for some kind of **staging** environment.
+
+Beside these two branches (`dev` and `master`) there are other three types of branches that will be explained in **Flow** section:
+- `feature` branches
+- `hotfix` branches
+- `release` branch
+
+## Flow
+
+Here is explained _Git branching model_ with branches management and releasing process.  
+
+### Feature branches
+
+Let's start with `feature` branches. They represent some **feature** needs to be implemented. Depending on which _project manager tool_ is used for project, they have internally agreed naming convention. One of the naming example would be - `feature-login-support` or `feature-21-login-support`. Every feature **must** start from `dev` branch and **must** be eventually merged to `dev` branch. We're achieving this via following commands:
+
+```
+git checkout -b feature-example dev
+git add .
+git commit -m "Implement example feature"
+git push origin feature-example
+```
+
+With this commands **feature** branch is created starting from `dev`, implemented that feature and pushed that to remote repository. **Pull request** must be created via UI, since there's no support for command line. When feature **passes tests**, is **approved** and **merged**, we **STRONGLY recommend** to delete branch by typing following commands:
+
+```
+git checkout dev
+git pull origin dev --rebase
+git branch -d feature-example
+```
+
+This will keep local repository clean.
+
+Just to be clear, when richer semantic in flow is needed, **feature** branch can be started with another words such as `feature-*`, `task-*`, `bug-*`, `fix-*`, etc. It's useful because branch names contain type of task that we've worked on.
+
+### Hotfix branches
+
+This branch must be created when there's some **critical** bug in **production**. This branch is created from `master` branch should be merged to `master` branch. Also, **hotfix** branch must be merged into `dev` branch.
+
+### Release branch
+
+**Release** branch is used only to "put" code from `dev` to `master`. Complete process of releasing process is given in section [Releasing Process](https://github.com/lambdaworkshub/contributing-guide/blob/master/git/Releasing%20Process.md). We just want to point you for **tagging**.
```

Comment:
This is not the only purpose of **release** branches.

Reply:
Give an example.